proxy: Re-enable proxy rule installation in tunnel mode

sayboras opened the pull request:
### Description

This commit is to re-enable proxy rule installation in tunnel mode, as
route 2005 was added back, and we need this rule to handle the
hairpinning trafic in Ingress L7 proxy if the backend is on the same
node.

Relates: 0ebe5162373c00f85e7ae43d0bc5d474fa08c485
Relates: https://github.com/cilium/cilium/pull/29530, https://github.com/cilium/cilium/issues/29864

Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
proxy: Re-enable proxy rule installation in tunnel mode

This commit is to re-enable proxy rule installation in tunnel mode, as
route 2005 was added back, and we need this rule to handle the
hairpinning trafic in Ingress L7 proxy if the backend is on the same
node.

Relates: 0ebe5162373c00f85e7ae43d0bc5d474fa08c485
Relates: #29530, #29864

Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
ci-gke: Add test for hair-pinning Ingress case

Signed-off-by: Tam Mach <tam.mach@cilium.io>

julianwiedmann commented on pkg/proxy/proxy.go:
```diff
@@ -412,7 +412,7 @@ func (p *Proxy) ReinstallRoutingRules() error {
+		if !option.Config.EnableIPSec {
 			if err := removeFromProxyRoutesIPv4(); err != nil {
 				return err
 			}
```
It might make sense to refactor this into:

```
if (foo)
	installFromProxyRoutes()
else
	removeFromProxyRoutes()
```

But I believe you actually need the routes regardless of whether IPsec is enabled, correct?

sayboras replied:
Yes, that is correct, just a small note that IPSec + Ingress/Gateway API are not compatible due to KPR/nodeport requirement.

julianwiedmann replied:
right! So for the time being, should we spell this out as

```
if (Ingress || IPsec with native-routing)
	installFromProxyRoutes()
else
	removeFromProxyRoutes()
```

(if there's no easy way to express "Ingress enabled", then let's say "`!IPsec || IPsec with native-routing`" instead?)

sayboras committed:
clustermesh: Enable legacy host routing for Ingress

After #2206, BPF host routing is enabled by default, however, this drops
the reply packet to proxy. This commit is to enable legacy host routing
as a workaround.

Relates: #31653, #22006

Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
temp commit for testing

Signed-off-by: Tam Mach <tam.mach@cilium.io>
